src/main/java/io/strimzi/kafka/bridge/SinkBridgeEndpoint.java

```diff
@@ -232,86 +215,71 @@ protected void assign(Handler<AsyncResult<Void>> assignHandler) {
+     *
+     * @return records polled from the Kafka cluster
      */
-    private void setPartitionsAssignmentHandlers() {
-        this.consumer.partitionsRevokedHandler(partitions -> {
-
-            log.debug("Partitions revoked {}", partitions.size());
-
-            if (log.isDebugEnabled() && !partitions.isEmpty()) {
-                for (TopicPartition partition : partitions) {
-                    log.debug("topic {} partition {}", partition.getTopic(), partition.getPartition());
-                }
-            }
-
-            if (this.partitionsAssignmentHandle != null) {
-                this.partitionsAssignmentHandle.handleRevokedPartitions(partitions);
-            }
-        });
-
-        this.consumer.partitionsAssignedHandler(partitions -> {
-
-            log.debug("Partitions assigned {}", partitions.size());
-
-            if (log.isDebugEnabled() && !partitions.isEmpty()) {
-                for (TopicPartition partition : partitions) {
-                    log.debug("topic {} partition {}", partition.getTopic(), partition.getPartition());
-                }
-            }
-
-            if (this.partitionsAssignmentHandle != null) {
-                this.partitionsAssignmentHandle.handleAssignedPartitions(partitions);
-            }
-        });
-    }
-
-    protected void consume(Handler<AsyncResult<KafkaConsumerRecords<K, V>>> consumeHandler) {
-        this.consumer.poll(Duration.ofMillis(this.pollTimeOut), consumeHandler);
+    protected ConsumerRecords<K, V> consume() {
+        log.trace("Poll thread {}", Thread.currentThread());
+        return this.consumer.poll(Duration.ofMillis(this.pollTimeOut));
     }
 
-    protected void commit(Map<TopicPartition, io.vertx.kafka.client.consumer.OffsetAndMetadata> offsetsData,
-        Handler<AsyncResult<Map<TopicPartition, io.vertx.kafka.client.consumer.OffsetAndMetadata>>> commitOffsetsHandler) {
-        this.consumer.commit(offsetsData, commitOffsetsHandler);
+    /**
+     * Commit the offsets on topic partitions provided as parameter
+     *
+     * @param offsetsData map containing topic partitions and corresponding offsets to commit
+     * @return map containing topic partitions and corresponding committed offsets
+     */
+    protected Map<TopicPartition, OffsetAndMetadata> commit(Map<TopicPartition, OffsetAndMetadata> offsetsData) {
+        log.trace("Commit thread {}", Thread.currentThread());
+        // TODO: doesn't it make sense to change using the commitAsync?
+        //       does it still make sense to return the offsets we get as parameter?
+        this.consumer.commitSync(offsetsData);
+        return offsetsData;
     }
 
-    protected void commit(Handler<AsyncResult<Void>> commitHandler) {
-        this.consumer.commit(commitHandler);
+    /**
+     * Commit offsets returned on the last poll() for all the subscribed list of topics and partitions
+     */
+    protected void commit() {
```

Comment:
Maybe it would be clearer if this one was called `commitLastPolledOffsets()`?

Reply:
no hurt to change that even if `commit` is just the same name as used in the Java Kafka client having the same behaviour.